UI/importers/classic.cpp

```diff
@@ -0,0 +1,586 @@
+	    first[0] == '/') {
+		if (out.find(c_name) != out.end()) {
+			Json::array arr = out[c_name].array_items();
+			if (out[c_name].is_string()) {
+				Json::array new_arr = Json::array{};
+				string str = out[c_name].string_value();
+				create_string_obj(str, new_arr);
+				arr = new_arr;
+			}
+
+			create_string_obj(first, arr);
+			out[c_name] = arr;
+		} else {
+			create_string(c_name, out, first);
+		}
+	} else if (first[0] == '"') {
+		string str = first.substr(1, first.size() - 2);
+
+		if (out.find(c_name) != out.end()) {
+			Json::array arr = out[c_name].array_items();
+			if (out[c_name].is_string()) {
+				Json::array new_arr = Json::array{};
+				string str1 = out[c_name].string_value();
+				create_string_obj(str1, new_arr);
+				arr = new_arr;
+			}
+
+			create_string_obj(str, arr);
+			out[c_name] = arr;
+		} else {
+			create_string(c_name, out, str);
+		}
+	} else if (first.find('.') != string::npos) {
+		create_double(c_name, out, first);
+	} else {
+		create_int(c_name, out, first);
+	}
+}
+
+static Json::object create_object(Json::object &out, string &line, string &src);
+
+static Json::array create_sources(Json::object &out, string &line, string &src)
+{
+	Json::array res = Json::array{};
+
+	line = ReadLine(src);
+	size_t l_len = line.size();
+	while (line != "" && line[l_len - 1] != '}') {
+		size_t end_pos = line.find(':');
+
+		if (end_pos == string::npos)
+			return Json::array{};
+
+		size_t start_pos = 0;
+		while (line[start_pos] == ' ')
+			start_pos++;
+
+		string name = line.substr(start_pos, end_pos - start_pos - 1);
+
+		Json::object nul = Json::object();
+
+		Json::object source = create_object(nul, line, src);
+		source["name"] = name;
+		res.push_back(source);
+
+		line = ReadLine(src);
+		l_len = line.size();
+	}
+
+	if (!out.empty())
+		out["sources"] = res;
+
+	return res;
+}
+
+static Json::object create_object(Json::object &out, string &line, string &src)
+{
+	size_t end_pos = line.find(':');
+
+	if (end_pos == string::npos)
+		return Json::object{};
+
+	size_t start_pos = 0;
+	while (line[start_pos] == ' ')
+		start_pos++;
+
+	string name = line.substr(start_pos, end_pos - start_pos - 1);
+
+	Json::object res = Json::object{};
+
+	line = ReadLine(src);
+
+	size_t l_len = line.size() - 1;
+
+	while (line != "" && line[l_len] != '}') {
+		start_pos = 0;
+		while (line[start_pos] == ' ')
+			start_pos++;
+
+		if (line.substr(start_pos, 7) == "sources")
+			create_sources(res, line, src);
+		else if (line[l_len] == '{')
+			create_object(res, line, src);
+		else
+			create_data_item(res, line);
+
+		line = ReadLine(src);
+		l_len = line.size() - 1;
+	}
+
+	if (!out.empty())
+		out[name] = res;
+
+	return res;
+}
+
+string ClassicImporter::Name(const string &path)
+{
+	return GetFilenameFromPath(path);
+}
+
+int ClassicImporter::ImportScenes(const string &path, string &name, Json &res)
+{
+	char *file_data = os_quick_read_utf8_file(path.c_str());
```

Comment:
You can `#include <util/util.hpp>`, then you can use `BPtr<char>` with the `file_data` value here to make it automatically free itself with RAII rather than having to explicitly call `bfree()`  on it.  You can basically get rid of the `bfree()` call with it.  The `if (!file_data)` will also still work fine so you don't have to worry about that.